### PR TITLE
feat(docker): use docker-compose to run integration tests for E2E tests support

### PR DESCRIPTION
**Pull request by Blackweather**
* add new target for docker-compose
* allow running docker-compose E2E test with pattern docker-compose.$SOME_STRING.ci.y(a)ml

### CODE REVIEW

File: src/AM.Condo/Targets/Test/DockerCompose.targets [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xlmns="http://schemas.microsoft.com/developer/msbuild/2003">
+	<Target Name="DockerComposeTest" Condition=" $(DockerRequired) AND $(DOCKER_COMPOSE_PATHS) != '' ">
+        <!-- .sh file needs to be created to be run directly by bash with #!/bin/bash instead of dash used by agent -->
+        <!-- Init bash script in debug mode -->
+		<Exec Command="echo &apos;#!/bin/bash -x&apos; > compose.sh" />
```
Comment (sjk07) [inline src/AM.Condo/Targets/Test/DockerCompose.targets]: What and where is this compose.sh file?
why are we running in debug mode?

File: src/AM.Condo/Targets/Test/DockerCompose.targets [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xlmns="http://schemas.microsoft.com/developer/msbuild/2003">
+	<Target Name="DockerComposeTest" Condition=" $(DockerRequired) AND $(DOCKER_COMPOSE_PATHS) != '' ">
```
Comment (sjk07) [inline src/AM.Condo/Targets/Test/DockerCompose.targets]: If we want to use DOCKER_COMPOSE_PATHS as an override then it should be created as a prepare step. 

We also need the discovery of dockercompose.test files

**Comment by sjk07** [comment]
Hi @Blackweather! thanks for the contribution. We have an open issue for this issue you are trying to work on. #49 

Condo is a build system that will do discovery and find the files necessary to execute scenarios. It does this by having an opinionated file / filename structure. I do like the first pass of using a env var to inject the paths I would much more prefer following the discovery pattern stated in #49

File: src/AM.Condo/Targets/Test/DockerCompose.targets (hunk 1 above)
Comment (Blackweather) [reply src/AM.Condo/Targets/Test/DockerCompose.targets]: compose.sh is temporarily being created in the current working directory and later deleted (line 18)

Comment (Blackweather) [reply src/AM.Condo/Targets/Test/DockerCompose.targets]: We are running in debug mode so that the .sh script outputs the performed commands

Comment (Blackweather) [reply src/AM.Condo/Targets/Test/DockerCompose.targets]: simplified call, there's no need to run bash scripts

File: src/AM.Condo/Targets/Test/DockerCompose.targets (hunk 2 above)
Comment (Blackweather) [reply src/AM.Condo/Targets/Test/DockerCompose.targets]: override is not needed anymore, we are using discovery pattern with DockerRoot path

**Comment by Blackweather** [comment]
@sjk07 I applied your suggestions: using Discovery pattern and avoiding overrides. 

However I would like to propose new filename pattern for E2E tests (docker-compose.$SOME_STRING.ci.y(a)ml) for following reasons:
* to avoid accidentally launching tasks that are designed for local operations
* docker-compose that should run on CI needs some additional setup, for example build number pattern in docker images which we don't use locally (build number is generated after CI build)

**Comment by sjk07** [comment]
@Blackweather 

> to avoid accidentally launching tasks that are designed for local operations

Can you give me a scenario where this would happen, what local operations would not be suitable in a CI env?

> docker-compose that should run on CI needs some additional setup, for example build number pattern in docker images which we don't use locally (build number is generated after CI build)

Im not sure of any configuration you would need to setup.
when CI builds the images they will already be in the docker cache, if you do not specify the tag it will take the latest built image (that should the images you want to test).

**Comment by sjk07** [comment]
no progress

**Pull request closed without merging**